### PR TITLE
Adds Hilt to the project

### PR DESCRIPTION
**Pull request by manuelvicnt**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -120,6 +106,21 @@ dependencies {
     implementation "androidx.navigation:navigation-fragment-ktx:$navigationVersion"
     implementation "androidx.navigation:navigation-ui-ktx:$navigationVersion"
 
+    // MultiDex
+    implementation "androidx.multidex:multidex:$multiDexVersion"
```
Comment (JoseAlcerreca) [inline app/build.gradle:110]: Ouch, this is bad. Do you have stats?

File: app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/AppNavigationTest.kt [hunk 2]
```diff
@@ -57,21 +60,22 @@ import org.junit.runner.RunWith
...
 
     @Before
     fun init() {
-        tasksRepository = ServiceLocator.provideTasksRepository(getApplicationContext())
-    }
-
-    @After
-    fun reset() {
-        ServiceLocator.resetRepository()
+        hiltRule.inject()
```
Comment (JoseAlcerreca) [inline app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/AppNavigationTest.kt:78]: This is new to me, add comment explaining why it's needed?

File: app/src/debug/java/com/example/android/architecture/blueprints/todoapp/TestHiltActivity.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * limitations under the License.
+ */
+
+package com.example.android.architecture.blueprints.todoapp
+
+import androidx.appcompat.app.AppCompatActivity
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class TestHiltActivity : AppCompatActivity()
```
Comment (JoseAlcerreca) [inline app/src/debug/java/com/example/android/architecture/blueprints/todoapp/TestHiltActivity.kt:23]: nit: empty line at EOF

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt [hunk 4]
```diff
@@ -27,12 +27,9 @@ import timber.log.Timber.DebugTree
  *
  * Also, sets up Timber in the DEBUG BuildConfig. Read Timber's documentation for production setups.
  */
+@HiltAndroidApp
```
Comment (JoseAlcerreca) [inline app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt:30]: Update kdoc

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/di/AppModule.kt [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import javax.inject.Qualifier
+import javax.inject.Singleton
+import kotlin.annotation.AnnotationRetention.RUNTIME
+
+@InstallIn(ApplicationComponent::class)
+@Module
+object AppModule {
```
Comment (JoseAlcerreca) [inline app/src/main/java/com/example/android/architecture/blueprints/todoapp/di/AppModule.kt:40]: Add kdoc to these two modules, explaining why they have to be separate.

File: app/build.gradle (hunk 1 above)
Comment (manuelvicnt) [reply app/build.gradle:110]: As talked offline, this is only happening on `androidTest`. Updating minSdk 21 to temporary fix it

File: app/src/debug/java/com/example/android/architecture/blueprints/todoapp/TestHiltActivity.kt (hunk 3 above)
Comment (manuelvicnt) [reply app/src/debug/java/com/example/android/architecture/blueprints/todoapp/TestHiltActivity.kt:23]: Fixed, thanks!

File: app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/AppNavigationTest.kt (hunk 2 above)
Comment (manuelvicnt) [reply app/src/androidTest/java/com/example/android/architecture/blueprints/todoapp/AppNavigationTest.kt:78]: Fixed. Thanks!

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt (hunk 4 above)
Comment (manuelvicnt) [reply app/src/main/java/com/example/android/architecture/blueprints/todoapp/TodoApplication.kt:30]: Fixed, thanks!

File: app/src/main/java/com/example/android/architecture/blueprints/todoapp/di/AppModule.kt (hunk 5 above)
Comment (manuelvicnt) [reply app/src/main/java/com/example/android/architecture/blueprints/todoapp/di/AppModule.kt:40]: Done and also added kdoc to the test module

**Pull request merged**